llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -938,9 +938,9 @@ bool RISCVVLOptimizer::checkUsers(const MachineOperand *&CommonVL,
     // The SEW and LMUL of destination and source registers need to match.
 
     // We know that MI DEF is a vector register, because that was the guard
-    // to call this function.
-    assert(isVectorRegClass(UserMI.getOperand(0).getReg(), MRI) &&
-           "Expected DEF and USE to be vector registers");
+    // to call this function, so we don't need to assert it.
+    assert(isVectorRegClass(UserOp.getReg(), MRI) &&
+           "Expected consumed operand to be a vector register");
```

Comment:
Is UserOp.getReg() the same as MI.getOperand(0).getReg()? Which is what's guarded in the callsite at tryReduceVL? 

Should we maybe just remove this assertion entirely?

Reply:
We can remove it entirely.